### PR TITLE
community: Add Fireworks callback handler to track token usage info

### PR DESCRIPTION
**Pull request by red40maxxer**
- **Description:** I implemented `FireworksCallbackHandler` using `OpenAICallbackHandler` as a reference, it allows you to access token usage information in the same way.
- **Issue:** None
- **Dependencies:** None 

Usage example:
```
from langchain.llms.fireworks import Fireworks
from langchain_community.callbacks import get_fireworks_callback
llm = Fireworks()
with get_fireworks_callback() as cb:
    result = llm("Hello world!")
    print(cb)
```

### CODE REVIEW

**Comment by mad-cat-lon** [comment]
@efriis Would you mind approving these workflows? Thanks

**Comment by mad-cat-lon** [comment]
Hey @baskaryan could I get you or another maintainer to take a look at this PR? Would be super useful to have as I need to compare token costs between OpenAI and Fireworks for my eval pipeline, thank you! :)

**Comment by baskaryan** [comment]
cc @hinthornw

**Comment by hinthornw** [comment]
Could we add this to ChatFireworks as well before landing?

File: libs/community/langchain_community/callbacks/fireworks_callback.py [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    "stability/japanese-stablelm-instruct-beta-70b": (0.7, 2.8),
+    "stability/japanese-stablelm-instruct-gamma-7b": (0.2, 0.8),
+    "stability/japanese-stable-vlm": (0.7, 2.8),
+    "fireworks/llama-v2-7b": (0.2, 0.8),
+    "fireworks/llava-codellama-34b": (0.7, 2.8),
+    "fireworks/llava-v15-13b-fireworks": (0.2, 0.8),
+    "fireworks/mistral-7b": (0.2, 0.8),
+    "fireworks/mixtral-8x7b": (0.4, 1.6),
+    "stability/stablelm-zephyr-3b": (0.2, 0.8),
+    "fireworks/zephyr-7b-beta": (0.2, 0.8),
```
Comment (benjibc) [inline libs/community/langchain_community/callbacks/fireworks_callback.py:38]: we have flat pricing now https://fireworks.ai/pricing

**Review by benjibc — COMMENTED** [review]
lgtm, not familiar with the handler code, but pricing needs to be updated

**Comment by ccurme** [comment]
Closing but tag me and I will re-open if anyone wants to push this with the following updates:

- Correct pricing information;
- Reads token counts off `AIMessage.usage_metadata`, which `ChatFireworks` outputs for both [invoke](https://github.com/langchain-ai/langchain/blob/37f5ba416ebc839035aaa9fc5513143982629090/libs/partners/fireworks/langchain_fireworks/chat_models.py#L526-L530) and [streaming](https://github.com/langchain-ai/langchain/blob/37f5ba416ebc839035aaa9fc5513143982629090/libs/partners/fireworks/langchain_fireworks/chat_models.py#L244-L249).

Should be tested (at least locally) using `ChatFireworks` in [langchain-fireworks](https://github.com/langchain-ai/langchain/tree/master/libs/partners/fireworks) library.

See example with OpenAI callback [here](https://github.com/langchain-ai/langchain/blob/37f5ba416ebc839035aaa9fc5513143982629090/libs/community/langchain_community/callbacks/openai_info.py#L224).

**Pull request closed without merging**